tools/toolchains/__init__.py

```diff
@@ -792,7 +792,7 @@ def parse_dependencies(self, dep_path):
         """Take in a dependency file generated by the compiler and build a list of
            all files that the dep_path depends on.
         """
-        pass
+        raise NotImplemented
```

Comment:
Does this make sense? You can't instantiate meta class - ``TypeError``, same is true for the concrete classes, therefore this never can get called ? I would expect a raise without a use of abc meta class